### ISSUE TITLE
Provide guidance for cross-browser testing

### ISSUE THREAD

**Issue by foolip** [google/WebFundamentals#6952]
**Page Affected:** https://developers.google.com/web/tools/

https://github.com/google/WebFundamentals/pull/5644 added this documention on Puppeteer to /web/tools/. (It's great, I've used Puppeteer for some scraping tasks and I love it. @Hexcles also found it great in https://github.com/Hexcles/wpt-puppeteer, an experiment in using it for browser testing.)

**What needs to be done?**

Puppeteer only supports Chrome, and we should give clear guidance to web developers on current best practices for cross-browser testing. Libraries using WebDriver are currently the only practical option, and https://www.seleniumhq.org/ are the most popular, but there's also https://webdriver.io/ and probably many others.

Main options:
- A new entry for WebDriver with an entry in /web/tools/ and cross-linking with Puppeteer.
- A single page for test automation that lays out what you need to use for which use case.

@ebidel @aslushnikov WDYT?

**Comment by foolip** [comment]
@AutomatedTester originally pointed this out to me.

**Comment by aslushnikov** [comment]
> Puppeteer only supports Chrome

There's now an experimental [puppeteer-firefox](http://npmjs.com/package/puppeteer-firefox): https://github.com/GoogleChrome/puppeteer/pull/3628

**Comment by foolip** [comment]
That is very cool, glad to see progress being made on that!

@LukeZielinski and @mdittmer will be working on https://github.com/WICG/devtools-protocol in the coming year and whether by that route or more Puppeteer backends in the style of puppeteer-firefox, I hope and believe we'll eventually get to Puppeteer supporting all browsers that web developers need to test on.

The timeline is a bit unclear, of course. In the interim, what guidance do you think we should give to web developers wanting to test their sites cross-browser?

**Comment by ebidel** [comment]
d.g.c/web/tools is meant to cover Chrome/Google's offerings in the tooling space. I'm not it's the right place to highlight best practices across many other solutions. IMO, the space is also covered fairly well outside of our own documentation and I'm not sure similar content would bring to the table.

That said,  web.dev is probably a better place to talk about cross-browser solutions. We've already got some great content there that covers community frameworks, libs, plugins, and tooling.

**Comment by TechQuery** [comment]
@aslushnikov I don't think it is a good idea to bind **Puppeteer API** to _a specific version of Firefox with patches_.

@nylen I'm developing [`puppeteer-ie` package](https://www.npmjs.com/package/puppeteer-ie).

**Comment by ebidel** [comment]
I'm not sure we're _that_ organized :) Generally, anything on https://developers.google.com/web/tools/ are tools/libs developed and maintained by our developer relations team and/or the Chrome devtools teams within Chrome. The code for these projects mostly lives under https://github.com/GoogleChrome or https://github.com/GoogleChromeLabs. ChromeDriver is an older project, lives in other repos, and is maintained by other teams within chromium. 

Puppeteer for FF has the same API as the chromium-based version so I'm not sure anything would need to change about it's location. If the API diverges in the future, we can include ways for users to switch between browser variants in the docs, similar to the language selectors at the top of https://cloud.google.com/appengine/docs/flexible/nodejs/quickstart.

FWIW, I think we can and should do a whole lot more with mentioning and linking other tooling options on https://developers.google.com/web/tools/. What I'm hesitant about is having folks write new documentation, articles, and tutorials for other tools which we don't own or are experts in. Those resources are better left to the communities that build and maintain those projects. 
The other issues is one that @nylen pointed out,...there's basically a new lib/framework/tool every few weeks! How do we chose what goes on https://developers.google.com/web/tools? Etc.

Does that make sense?

**Comment by AutomatedTester** [comment]
> How do we chose what goes on https://developers.google.com/web/tools? 

What tools add the open web and don't push a "Chromium only" perception of the web but I am a Mozillian so my bias is around giving people the choice of browser. So pick tools that enable the open web :)

Since you've clarified with 
> Generally, anything on https://developers.google.com/web/tools/ are tools/libs developed and maintained by our developer relations team and/or the Chrome devtools teams within Chrome. 

the keys and responsibility are in your hands.
